abcclassroom/dirsetup.py

```diff
@@ -0,0 +1,94 @@
+"""
+abc-classroom.dir-setup
+========================
+
+"""
+
+import os
+import shutil
+import argparse
+import abcclassroom
+
+
+def path_to_example(dataset):
+    """
+    Gets the path to an example dataset.
+
+    Parameters
+    ----------
+    dataset : string
+        stringy boi
+    """
+    abcclassroom_path = os.path.split(abcclassroom.__file__)[0]
+    data_dir = os.path.join(abcclassroom_path, "example-data")
+    data_files = os.listdir(data_dir)
+    if dataset not in data_files:
+        raise KeyError(dataset + " not found in abc-classroom example data.")
+    return os.path.join(data_dir, dataset)
+
+
+def directory_setup():
+    """
+    Construct a file path to an example dataset.
+    This file defines helper functions to access data files in this directory,
+    to support examples. Adapted from the PySAL package.
+
+    Parameters
+    ----------
+    dataset: string
+        Name of a dataset to access (e.g., "epsg.json", or "RGB.byte.tif")
+
+    Returns
+    -------
+        A file path (string) to the dataset
+    """
+    # Creating the default names for the cloned and template repos.
+    cloned = "cloned_repos"
+    template = "template_repos"
+    course = "course_dir"
+    # Making sure the configuration file is where it's supposed to be.
+    config = path_to_example("sample_config.yml")
+    if not os.path.exists(config):
+        raise ValueError(
+            "Configuration file can't be located, please ensure abc-classroom has been installed correctly"
+        )
+    # Allows users to rename the cloned and template repos.
+    parser = argparse.ArgumentParser(description=directory_setup.__doc__)
```

Comment:
@nkorinek this is where i got a bit confused. this is a function called directory_setup and you have a `directory_setup.__doc__` call. perhaps you can explain what this does but also let's just come up with a better naming convention!!

@kcranston what would you call these functions?

Reply:
In description=<name> , name should be the function name (which I assume is now "quickstart"). That construct simply uses the docstring as the description when printing the help message on the cmdline.